docs/proposals/resource-catalog.md

```diff
@@ -0,0 +1,222 @@
+	// Generation indicates which level of the Entry was accepted
+	Generation int
+
+	// Description is client controlled.  It describes the resources being provided.
+	Description string
+}
+```
+
+## ResourceCatalogEntry
+ResourceCatalogEntry is namespace scoped resources.  Its labels must match the selector on a ResourceCatalog
+to get included in one or more Catalogs.  
+The Spec contains a "must pick one" set of pointers to different kinds of "claim satisfiers".  The built-in
+ones will be controller managed for later claim resource instantiation.  I've added two as a for instance,
+but you could pretty easily envision a remote endpoint and a PodSpec sort of satisfier as well.
+The RESTStorage for a ResourceCatalogEntry is pretty complicated.  In the case
+of a LocalResourcSpec, it must locate all resources that will be exposed and run secondary ACL checks to be 
+sure that the creator has GET rights on the resources in question.  Then the RESTStorage itself creates the 
+Status before persisting.  If you don't do it in the RESTStorage, you'll lose track of who requested it and
+you'll be unable to run the proper ACL checks.
+
+```go
+// ResourceCatalogEntry is a namespace scoped resource.  In order to be added to a ResourceCatalog the labels
+// on the Entry must match the selector of the Catalog
+type ResourceCatalogEntry struct {
+	unversioned.TypeMeta
+	api.ObjectMeta
+
+	Spec   ResourceCatalogEntrySpec
+	Status ResourceCatalogEntryStatus
+}
+
+// ResourceCatalogEntrySpec gives information about the Entry and describes how a claim would be satisfied.
+// Only one of the pointers maybe specified.  Only the Description is exposed.
+// An Update can be used to re-detect the exposed resources.  This could ripple through and kick synchronization
+// across all claims.  Otherwise, new claims get new resource and old ones keep their existing ones.
+type ResourceCatalogEntrySpec struct {
+	// Description says what this does
+	Description string
+
+	// LocalResources describes the case where resources exist in this namespace that will be expose via a claim
+	// to another namespace
+	LocalResources *LocalResourceSpec
+
+	// ResourceList describes the case where we have list that we want to create in the claim-namespace, but the
+	// resources don't actually exist in this namespace
+	ResourceList *ResourceListSpec
+
+	// Extension is a spot for a generic spec.  This would need a custom controller to make use of.  Imagine this
+	// as your placeholder for an openshift template.
+	Extension runtime.Object
+}
+
+// LocalResourceSpec describes the case where resources exist in this namespace that will be expose via a claim
+// to another namespace.
+type LocalResourceSpec struct {
+	// Transitive indicates whether "depends-on" annotations in the linked objects are chased and added to Status
+	Transitive bool
+
+	// ExposedResources is the list of resources (name or name/UID) to be included in the exposure
+	ExposedResources []api.LocalObjectReference
```

Comment:
For this kind of provisioning (copy cool stuff from my namespace), I think the naming should match except for the claimant.   I think that tweaking object references will be the hardest part and may force people to templates of we don't handle common cases well.